feat(docs): add performance bench and tuning pages (#2689)
kestra-io/docs: content/docs/performance/benchmark.md

| 750	| 1500	| 0,26 |
| 1000	| 2000	| 0,3 |
| 1250	| 2500	| 0,3 |
| 1500	| 3000	| 0,32 |
| 2000	| 4000	| 0,57 |
| 2500 |	5000	| 27 |

With medium messages of 16k.

![Kestra EE - Benchmark04 - Meidum messages](/docs/performance/bench04-EE-medium.png "Kestra OSS Benchmark04 with medium messages results")

| Executions(per minutes)	| Tasks (per minutes) |	Execution Latency (in seconds)|
|:--|:--|:--|
| 500	| 1000	| 0,26 |
| 750	| 1500	| 0,27 |
| 1000	| 2000	| 0,41 |
| 1250	| 2500	| 0,34 |
| 1500	| 3000	| 8 |

With big messages of 160k.

![Kestra EE - Benchmark04 - Big messages](/docs/performance/bench04-OSS-big.png "Kestra OSS Benchmark04 with big messages results")

| Executions(per minutes)	| Tasks (per minutes) |	Execution Latency (in seconds)|
|:--|:--|:--|
| 250	| 500	| 0,31 |
| 375	| 750	| 0,33 |
| 500	| 1000	| 0,48 |
| 625	| 1250	| 104 |

**Key takeaways**
- With small messages, we have approximately the same results as the one of the first benchmark which is expected.
- With medium messages, Kestra can sustain up to 1250 executions or 2500 tasks per minute with an execution duration of less than 1s which is what we could realistically target for such workflow.

> - With big messages, performance starts to degrade vastly which is expected as the Worker would have more work to do, and the Executor is sensible to the size of the execution message.

Reviewer: ```suggestion
- With big messages, performance starts to degrade vastly, which is expected as the Worker would have more work to do, and the Executor is sensitive to the size of the execution message.
```

> - Realtime triggers will create executions at a high rate, we see here that Kestra EE particularly shine in such scenario as it could sustain way more tasks per minutes (4000) than Kestra OSS (2500) for small messages.

Reviewer: ```suggestion
- Real-time triggers will create executions at a high rate. Here, Kestra EE particularly shines in such scenarios, as it could sustain way more tasks per minute (4000) than Kestra OSS (2500) for small messages.
```

## Conclusion

> Kestra is a platform not a framework, it comes bundled with logs, metrics, retry, SLA, error handling, a rich UI, governance, observability, ... This of course has a cost if you compare it with a framework or a lighter tool but we try to balance a rich features platform with good workflow execution performance.

Reviewer: ```suggestion
Kestra is a platform, not a framework. It comes bundled with logs, metrics, retry, SLA, error handling, a rich UI, governance, observability, etc. Of course, this has a cost if you compare it with a framework or a lighter tool, but we try to balance a rich features platform with good workflow execution performance.
```

> Kestra has been designed from the ground up with performance in mind, and the engineering team keep investing in performance optimization, this made Kestra one of the most performant workflow orchestrators.

Reviewer: ```suggestion
Kestra has been designed from the ground up with performance in mind, and the engineering team keeps investing in performance optimization, making it one of the most performant workflow orchestrators.
```

> Kestra performance focuses on workflow orchestration and task dispatching so your workflow would not loose time inside the orchestrator but spent most time inside your own workflow: the task execution.

Reviewer: ```suggestion
Kestra performance focuses on workflow orchestration and task dispatching, so your workflow will not lose time inside the orchestrator but spend most of its time inside your own workflow: task execution.
```

> Keep in mind that Kestra has been designed to scale horizontally, this means that you could expect Kestra's performance to increase nicely by adding more nodes (Executor or Worker) to your deployment.

Reviewer: ```suggestion
Keep in mind that Kestra has been designed to scale horizontally. This means that adding more nodes (Executor or Worker) to your deployment could increase Kestra's performance nicely.
```

::alert{type="info"}

> We will try to make our best to align this page with each new Kestra release.

Reviewer: ```suggestion
We will try our best to align this page with each new Kestra release.
```

> You can find details on how to run this benchmark by yourself in our [benchmarks](https://github.com/kestra-io/benchmarks) repository on GitHub.

Reviewer: ```suggestion
Details on how to run this benchmark yourself are in our [benchmarks](https://github.com/kestra-io/benchmarks) repository on GitHub.
```
